scipy/optimize/optimize.py

```diff
@@ -542,33 +551,34 @@ def fmin(func, x0, args=(), xtol=1e-4, ftol=1e-4, maxiter=None, maxfun=None,
            Griffiths and G.A. Watson (Eds.), Addison Wesley Longman,
            Harlow, UK, pp. 191-208.
 
+    .. [3] Gao, F. and Han, L. (2012), "Implementing the Nelder-Mead simplex
+           algorithm with adaptive parameters", Computational Optimization and
+           Applications, 51:1, pp. 259-277
+
     """
     opts = {'xatol': xtol,
             'fatol': ftol,
             'maxiter': maxiter,
             'maxfev': maxfun,
             'disp': disp,
             'return_all': retall,
-            'initial_simplex': initial_simplex}
+            'adaptive': adaptive, }
+    if initial_simplex is not None:
+        opts['initial_simplex'] = initial_simplex
 
     res = _minimize_neldermead(func, x0, args, callback=callback, **opts)
+    retlist = res['x'],
     if full_output:
-        retlist = res['x'], res['fun'], res['nit'], res['nfev'], res['status']
-        if retall:
-            retlist += (res['allvecs'], )
-        return retlist
-    else:
-        if retall:
-            return res['x'], res['allvecs']
-        else:
-            return res['x']
+        retlist += res['fun'], res['nit'], res['nfev'], res['status'],
+    if retall:
+        retlist += res['allvecs'],
+    return retlist
 
 
 def _minimize_neldermead(func, x0, args=(), callback=None,
                          maxiter=None, maxfev=None, disp=False,
-                         return_all=False, initial_simplex=None,
```

Comment:
Please keep the `initial_simplex` keyword. For backwards compatibility we need to retain it because it might be provided in a `minimize` options dict.